apps/04TowersOfHanoi.js

```diff
@@ -19,23 +19,42 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  // Function that moves the pieces
+  stacks[endStack].push(stacks[startStack].pop());
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(startStack, endStack) {
+  // This checks if the move you choose legal or not.
+  var startBlock = stacks[startStack][stacks[startStack].length - 1];
+  var endBlock = stacks[endStack][stacks[endStack].length - 1];
+  if (startBlock < endBlock || endBlock === undefined) {
+    movePiece(startStack, endStack);
```

Comment:
It works in this case, but at a higher level I don't think the `isLegal` function should be moving pieces around. `isLegal` is just a helper that looks at the move and decides if it's legal, it shouldn't actually be moving anything